pkg/providers/smi/provider.go

```diff
@@ -24,10 +25,15 @@ import (
 	listers "k8s.io/client-go/listers/core/v1"
 )
 
+// TCPPortMapper is capable of retrieving a TCP port mapping for a given service.
+type TCPPortMapper interface {
+	Find(svc k8s.ServiceWithPort) (int32, bool)
```

Comment:
Are there any issues with having this interface less restrictive?

Reply:
As answered [here](https://github.com/containous/maesh/pull/439/files#r383374288).